### PR TITLE
[API Refactor] Imperatively register bundles

### PR DESCRIPTION
**Pull request by VWoeltjen**
Addresses #450.

Opening PR early to solicit feedback from @akhenry and @larkin. 

Summary of changes:

* Add a `legacyRegistry` module, allowing for individual bundles to imperatively register the full contents of their `bundle.json` files from JavaScript code.
  * Currently, this just wires back into the existing framework layer code. As we update registration API (#462) we can do so "behind" this legacy adapter such that those changes occur in one place.
  * Created this in a new `src` folder; expect to add new registration-related classes in this hierarchy and to ultimately transition components into this hierarchy (per @larkin's [proposal](http://nasa.github.io/openmctweb/design/proposals/APIRedesign_PeteRichards.html#get-rid-of-views--representations--and-templates-) which, among other things, places specs and sources side-by-side.) Not sure when the natural time will be to start migrating things into here.
* Wrote and ran a script to rewrite `bundle.json` files as JavaScript files.
  * Included is explicit dependency on all `implementation` files, such that these get loaded sensibly as part of the AMD dependency graph.
  * Declared locations of third-party libraries used at the top-level `main.js`, to support the above
  * Made misc. changes to Angular bootstrapping (enabled strict DI, obeyed strict DI when decorating `$log` for logging levels). This allowed for verification of compatibility with a conventional `r.js` build/optimization step (not strictly a necessary outcome of this issue, but will want this to work in #460)

Still to-do:

- [x] Remove `bundles.json`
- [x] Add specs/tests new/changed files
- [x] Update test runner to remove `bundles.json` dependency, and to detect tests in new `src` hierarchy
- [x] Verify compatibility with dependent projects
- [x] Remove individual `bundle.json` files
- [x] Remove rebundling scripts (or move to a `scripts` folder or similar)

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Some specific questions:

* I'm feeling a little iffy about the approach to moving things into the `src` hierarchy. Not sure when in the process we should start moving things out of `platform` and into `src` (and, more importantly, when to move things out of `res` and `test` and make them peers.)
  * During #463 we'll hit the "components" topic, so that's the only obvious time. But this change is more ubiquitous, mainly due to the fact that specs will need to be relocated.
* The RequireJS config feels a little funny to me. 
  * Libraries which had previously been loaded per-bundle now get configured up in a top-level `main.js` (as a consequence of an ordering change; previously, bundle.json files were loaded, then RequireJS got configured, then implementations got loaded, whereas now implementations get loaded before bundle definitions are ever looked at). This feels like a hack but it's fairly consistent with what RequireJS docs instruct you to do.
  * The config necessary to get `legacyRegistry` available globally also felt a bit awkward. Curious if there's a better way to approach this. In particular, the fact that I need to depend on `src/BundleRegistry` instead of `./BundleRegistry` from `legacyRegistry` feels wrong to me. The [RequireJS docs](http://requirejs.org/docs/api.html#modulenotes) comment on this ("Relative module names are relative to other names, not paths") but the proposed solutions all seem like a poor fit.

**Comment by VWoeltjen** [comment]
Verified compatibility with dependent projects, with the caveat that `main.js` needs to be updated to reflect an appropriate set of active bundles at merge-time for each forked project. (Will need to address this in some fashion when splitting repositories after #460)

**Comment by VWoeltjen** [comment]
Think this is ready to merge at this point.

@larkin @akhenry - Be aware that this changes approach to running tests. `test.html` is no more (it depended on `bundles.json` which this change set gets rid of.) Instead, use `npm test` (also run as part of the Maven build.) I believe that all relevant documentation has been updated to reflect this.

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y &ast;
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

&ast; Some scripts have been added/changed in `platform/framework`, but tests there have only been updated to the point that they do not fail. Omitting tests here in anticipation of removal of that set of code entirely after changes to the way extensions are registered, #462

File: main.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define, window, requirejs*/
+
+requirejs.config({
+    "paths": {
```
Comment (akhenry) [inline main.js]: RequireJS seems to support appending config at 'runtime' - http://requirejs.org/docs/api.html#multiversion and http://stackoverflow.com/questions/18427457/adding-paths-to-requirejs-configuration-on-runtime 

Could we use this to push library dependencies into the bundles that require them?

Comment (akhenry) [inline main.js]: In this scenario, libraries that are globally required (such as angular) could remain in main.js.

Comment (VWoeltjen) [inline main.js]: Thanks, pushed updates which move non-global shim config into the modules where they are relevant

**Comment by akhenry** [comment]
Looks good.

Could we discuss the `src` directory briefly? What would our source tree ultimately look like after everything is migrated to `src/..`?
eg.
```
openmctweb
  |-src
  |  |-platform
  |  |  |-registration
  |  |  |  |-BundleRegistry.js
  |  |  |  |-BundleRegistrySpec.js
  |  |  |  |-legacyRegistry.js
  |  |  |  |-legacyRegistrySpec.js
  |  |  |-commonUI
  |  |  |  |-about
  |  |  |    |-about-dialog.html
  |  |  |    |-AboutController.js
  |  |  |    |-AboutControllerSpec.js
  |  |  |    ...
  |  |  |-containment
  |  |  ...
  |  |-examples
  |-docs

```

**Comment by akhenry** [comment]
Looks good.

### Reviewer Checklist

1. Changes appear to address issue? __Y__
2. Appropriate unit tests included? __Y__
3. Code style and in-line documentation are appropriate? __Y__
4. Commit messages meet standards? __Y__

**Pull request merged**